### DISCUSSION TITLE
How do I add fixed prefixes to dynamic topics

### DISCUSSION THREAD

**Discussion by kartikzheng** [fluent/fluent-bit#6096] [Q&A]
The logs collected by fluent-bit are output to Kafka. Can I specify a fixed prefix for topic? For example, I set the dynamic topic as a namespace, and I want to add a fixed prefix to the namespace. What should I do instead? Thank you very much.

`
[OUTPUT]

        Name           kafka

        Match          *

        Brokers        XXXX

        Topics         k8s.namespace

        Topic_key      namespace

        Dynamic_topic  on

        Timestamp_Key  @timestamp

        Retry_Limit    false
`

For example, add the fixed prefix "my"， Assume that namespace is test1, test2, and output topic is my_test1, my_test2.

**Comment by YE-Fan** [comment]
Expect this feature！

**Comment by agup006** [comment]
The easiest would be to add a filter right before, a pre-fix might be easy via Lua. A bit more complicated could be using the Nest Lift feature as well https://docs.fluentbit.io/manual/pipeline/filters/nest

**Reply by kartikzheng** [reply]
thanks. Nest Lift seems to only change the key, not the value
